src/test/java/org/jabref/logic/texparser/TexParserTest.java

```diff
@@ -0,0 +1,294 @@
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+
+import org.jabref.logic.importer.ImportFormatPreferences;
+import org.jabref.logic.importer.ParserResult;
+import org.jabref.logic.importer.fileformat.BibtexParser;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.texparser.Citation;
+import org.jabref.model.texparser.CrossingKeysResult;
+import org.jabref.model.texparser.TexParserResult;
+import org.jabref.model.util.DummyFileUpdateMonitor;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Answers;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+
+class TexParserTest {
+    private final String DARWIN = "Darwin1888";
+    private final String EINSTEIN = "Einstein1920";
+    private final String UNRESOLVED = "UnresolvedKey";
+    private ImportFormatPreferences importFormatPreferences;
+
+    @BeforeEach
+    void setUp() {
+        importFormatPreferences = mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS);
+    }
+
+    @AfterEach
+    void tearDown() {
+        importFormatPreferences = null;
+    }
+
+    private void testCite(String key, String citeString, boolean match) {
+        TexParserResult texParserResult = new DefaultTexParser().parse(citeString);
+        TexParserResult expectedResult = new TexParserResult();
+
+        if (match) {
+            expectedResult.getCitations().put(key, new ArrayList<>());
+            expectedResult.getCitations().get(key).add(
+                    new Citation(Paths.get("foo/bar"), 1, 0, citeString.length(), citeString));
+        }
+
+        assertEquals(expectedResult, texParserResult);
+    }
+
+    private void testMatchCite(String citeString) {
+        testCite(UNRESOLVED, citeString, true);
+    }
+
+    private void testNonMatchCite(String citeString) {
+        testCite(UNRESOLVED, citeString, false);
+    }
+
+    @Test
+    void testCiteCommands() {
+        testMatchCite("\\cite[pre][post]{UnresolvedKey}");
+        testMatchCite("\\cite*{UnresolvedKey}");
+        testMatchCite("\\parencite[post]{UnresolvedKey}");
+        testMatchCite("\\cite[pre][post]{UnresolvedKey}");
+        testMatchCite("\\citep{UnresolvedKey}");
+
+        testNonMatchCite("\\citet21312{123U123n123resolvedKey}");
+        testNonMatchCite("\\1cite[pr234e][post]{UnresolvedKey}");
+        testNonMatchCite("\\citep55{5}UnresolvedKey}");
+        testNonMatchCite("\\cit2et{UnresolvedKey}");
+    }
+
+    private void addCite(CrossingKeysResult expectedResult, String key, Path texFile, int line, int colStart, int colEnd, String lineText, boolean insert) {
```

Comment:
If you need to introduce helper methods in the test class, then this is usually a sign that the class under test needs to be refactored/extended to allow a better interaction.
For example, why not add a `addCitation(String key, Path file, ...)` method directly in `TexParserResult`?